x/market/keeper/deal.go

```diff
@@ -198,3 +198,19 @@ func (k Keeper) SetCurNumData(deal types.Deal) {
 	curNumData := deal.GetCurNumData() + 1
 	deal.CurNumData = curNumData
 }
+
+func (k Keeper) Verify(ctx sdk.Context, validatorAddr sdk.AccAddress, unSignedCert types.UnsignedDataValidationCertificate, cert types.DataValidationCertificate) (bool, error) {
```

Comment:
The `cert` has `unsignedCertificate` and `signature`, so you don't have to the `unSignedCert` argument.